packages/utils/src/createCacheAside.ts

```diff
@@ -0,0 +1,68 @@
+export type CacheClient<
+  TKey extends string,
+  TEntry extends { [K in TKey]: string },
+  TArgs extends [unknown],
+> = {
+  /**
+   * Get entries by their keys from cache.
+   */
+  get: (...args: TArgs) => Promise<TEntry[]>
+  /**
+   * Get products by their ids from cache.
+   */
+  set: (entries: TEntry[]) => Promise<unknown>
+}
+
+/**
+ * Creates a `Cache-Aside` callback.
+ * @param cache - The cache client. This will be used to get and set cached content.
+ * @param externalServiceCallback - The callback to be called when entries is not found in the
+ * cache.
+ * If the entries not found in `cache` gets returned from `externalServiceCallback`, they will be
+ * set in the cache and be appended to the result of the `cache.get` call.
+ *
+ * @see https://www.geeksforgeeks.org/cache-aside-pattern/
+ */
+export function createCacheAside<
+  TKey extends string,
+  TEntry extends { [K in TKey]: string },
+  TArgs extends [unknown],
+>(options: {
+  /**
+   * The cache client. This will be used to get and set cached content.
+   */
+  cache: CacheClient<TKey, TEntry, TArgs>
+  /**
+   * The callback to be called when entries is not found in the cache.
+   * If the entries not found in `cache` gets returned from `externalServiceCallback`, they will be
+   * set in the cache and be appended to the result from the cache.
+   */
+  fallback: (foundEntries: TEntry[], ...args: TArgs) => Promise<TEntry[]>
+}): (...args: TArgs) => Promise<TEntry[]> {
+  const { cache, fallback } = options
+
+  return async (...args) => {
+    // Get entries by keys from cache
+    const entries = await cache.get(...args).catch((e: unknown) => {
+      console.error(e)
+      console.error('Failed to get entries from cache. Se above error for more details')
+
+      return []
+    })
```

Comment:
question: shouldn't the cache client be responsible to catch exceptions here? Separation of concerns! 🪄 🧙‍♂️